sdk/storage/storage-file-share/review/storage-file-share.api.md

```diff
@@ -599,6 +599,7 @@ export class FileSystemAttributes {
 export interface FileUploadRangeFromURLOptionalParams extends coreHttp.RequestOptionsBase {
     sourceContentCrc64?: Uint8Array;
     sourceModifiedAccessConditions?: SourceModifiedAccessConditions;
+    sourceRange?: string;
```

Comment:
@jiacfan might have moved `sourceRange` out of the options bag to be a parameter.

Reply:
Any reason for it?

Reply:
Not sure. It was like that when this method was added in track 1.

Reply:
Source range is a required parameter, in track 1 we only involve optional parameters in optional parameter bag.

---
In reply to: [344427441](https://github.com/Azure/azure-sdk-for-js/pull/5980#discussion_r344427441) [](ancestors = 344427441)

Reply:
@jeremymeng, 
I think this is fine, in case we want to retain, we can add a transform like this - https://github.com/Azure/azure-sdk-for-js/pull/5980/commits/801dbcef04d0b13e42cdabfb15a2f80f744e981a.

Reply:
.NET and python have source range in the parameter list:
https://github.com/Azure/azure-sdk-for-net/blob/2e3a1eefaeefec1986f2fe258ffd5be050987e67/sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs#L1805

https://github.com/Azure/azure-sdk-for-python/blob/382994348b289d4cb90dc74e2ddc9fadc7c5824d/sdk/storage/azure-storage-file-share/azure/storage/fileshare/_file_client.py#L837

We should be consistent.

Reply:
The user-facing API is unchanged. 
https://github.com/Azure/azure-sdk-for-js/pull/5980/files/3c7cde92b986cae2939388cffb737b5a9721e27f#diff-458825853ee90eb4621dbbff6b5955f9R1290-R1296

```js
  public async uploadRangeFromURL(
    sourceURL: string,
    sourceOffset: number,
    destOffset: number,
    count: number,
    options: FileUploadRangeFromURLOptions = {}
  ): Promise<FileUploadRangeFromURLResponse> {
```
which is similar to the one in python -
```python
  def upload_range_from_url(self, source_url,
                              offset,
                              length,
                              source_offset,
                              **kwargs
                              ):
```

Reply:
`FileUploadRangeFromURLOptionalParams` is different from `FileUploadRangeFromURLOptions`.
- Public-facing `uploadRangeFromURL` function takes options param of type - `FileUploadRangeFromURLOptions`(which is not changed)
- Codegen `uploadRangeFromURL` function takes options param of type - 
`FileUploadRangeFromURLOptionalParams`(this is changed as in.. `sourceRange` parameter is moved to the options bag(`FileUploadRangeFromURLOptionalParams`))


Public-facing `uploadRangeFromURL` function internally calls the `uploadRangeFromURL` of codegen.
 https://github.com/Azure/azure-sdk-for-js/pull/5980/files/3c7cde92b986cae2939388cffb737b5a9721e27f#diff-458825853ee90eb4621dbbff6b5955f9R1310-R1318

Reply:
Ah got it now. I agree that it is fine if the public api is unchanged. @jiacfan what do you think?

Reply:
Agree.